lib/Parse/ParseDecl.cpp

```diff
@@ -2882,11 +2912,43 @@ ParserStatus Parser::parseNewDeclAttribute(DeclAttributes &Attributes,
 
     consumeAttributeLParen();
 
+    if (Tok.is(tok::code_complete)) {
+      if (CodeCompletionCallbacks) {
+        CodeCompletionCallbacks->completeDeclAttrParam(
+            ParameterizedDeclAttributeKind::AccessControl, 0, false);
+      }
+      consumeToken(tok::code_complete);
+      return makeParserCodeCompletionStatus();
+    }
+    
     // Parse the subject.
     if (Tok.isContextualKeyword("set")) {
       consumeToken();
     } else {
       diagnose(Loc, diag::attr_access_expected_set, AttrName);
+      
+      const Token &Tok2 = peekToken();
+      
+      if (Tok.is(tok::code_complete)) {
+        if (CodeCompletionCallbacks) {
+          CodeCompletionCallbacks->completeDeclAttrParam(
+              ParameterizedDeclAttributeKind::AccessControl, 0, false);
+        }
+        consumeToken(tok::code_complete);
+        return makeParserCodeCompletionStatus();
+      }
+      
+      if (Tok2.is(tok::code_complete) && Tok.is(tok::identifier) &&
+                 !Tok.isContextualDeclKeyword()) {
+        consumeToken(tok::identifier);
+        if (CodeCompletionCallbacks) {
+          CodeCompletionCallbacks->completeDeclAttrParam(
+              ParameterizedDeclAttributeKind::AccessControl, 0, false);
+        }
+        consumeToken(tok::code_complete);
+        return makeParserCodeCompletionStatus();
+      }
```

Comment:
Do you have a test case in mind that we need this code path for?

Reply:
This is mainly to complete `set` in cases like this (the identifier after "(" is meaningless to the parser):
```
private(se|
```
But not in cases like (the identifier after "(" is another contextual keyword):
```
private( nonisolated|
```

This is mainly to allow code completion triggered while in the middle of writing `set` not just directly after "(".

Reply:
See https://github.com/swiftlang/swift/pull/79974#discussion_r1994287469, I don’t think this is needed.